### PR TITLE
E1570: Refactoring TreeDisplayController

### PR DESCRIPTION
**Pull request by anushb**
Deployed on http://152.1.13.114:3000

### CODE REVIEW

**Comment by rapidwein** [comment]
Deployed on http://152.46.17.252:3000

File: app/controllers/menu_items_controller.rb [hunk 1]
```diff
@@ -154,13 +154,19 @@ def link
   str = params[:name]
   node = session[:menu].select(str)
   if node
-    redirect_to node.url
+    if node.id == 38   || node.id == 39 || node.id == 40  || node.id == 41 || node.id == 42 || node.id ==43 || node.id == 44 || node.id == 45 || node.id == 48 || node.id == 49
```
Comment (Winbobob) [inline app/controllers/menu_items_controller.rb]: Hardwired record numbers, should be retrieved from DB.

File: app/controllers/tree_display_controller.rb [hunk 2]
```diff
@@ -5,124 +5,53 @@ def action_allowed?
...
+      end
+
+      node_object = TreeFolder.find_by_name(name)
+      puts node_object.inspect
+      session[:root] = FolderNode.find_by_node_object_id(node_object.id).id
+      if node_object.name == "Courses"
+        session[:last_open_tab] = 1
+      elsif node_object.name == "Assignments"
+        session[:last_open_tab] = 2
+      elsif node_object.id ==1 ||node_object.id ==4 ||node_object.id ==5 ||node_object.id ==6 ||node_object.id ==7 ||node_object.id ==8 ||node_object.id ==9 ||node_object.id ==10 # if node_object_name is Questionnaires or its child_nodes
```
Comment (Winbobob) [inline app/controllers/tree_display_controller.rb]: Hardwired record numbers, should be retrieved from DB.

Comment (anushb) [inline app/controllers/tree_display_controller.rb]: Should we make the changes now? 

Thanks,
Anush

Comment (anushb) [inline app/controllers/tree_display_controller.rb]: Are we allowed to make the changes now? Or you were just informing us about how to do it in a correct way?

Thanks,
Anush

**Pull request closed without merging**